### PR TITLE
Allow multiline constructor arguments in an anonymous classes

### LINKED ISSUE #4289: Allow to multiline invoke constructor arguments in an anonymous class declaration
#### The PHP version you are using (`$ php -v`):
=> `7.2.14`

#### PHP CS Fixer version you are using (`$ php-cs-fixer -V`):
=> `2.14.0`

#### The command you use to run PHP CS Fixer:
=> `vendor/bin/php-cs-fixer fix --config .php_cs.dist --using-cache=yes`

#### The configuration file you are using, if any:
```php
<?php

$rules = [
    '@PSR2' => true,
];

$finder = PhpCsFixer\Finder::create()
    ->in(__DIR__ . '/config/container')
    ->in(__DIR__ . '/scripts')
    ->in(__DIR__ . '/tests')
;

return PhpCsFixer\Config::create()
    ->setRules($rules)
    ->setFinder($finder)
;
```

#### If applicable, please provide minimum samples of PHP code (as plain text, not screenshots):
 * before running PHP CS Fixer (no changes):
```php
<?php

class Foo
{
    public function foo($container): void
    {
        $bar = new class(
            $container->get('dependency1'),
            $container->get('dependency2'),
            $container->get('dependency3'),
            $container->get('dependency4'),
            $container->get('dependency5')
        ) extends Bar {
        };
    }
}
```

 * with unexpected changes applied when running PHP CS Fixer:
```php
<?php

class Foo
{
    public function foo($container): void
    {
        $bar = new class($container->get('dependency1'), $container->get('dependency2'), $container->get('dependency3'), $container->get('dependency4'), $container->get('dependency5')) extends Bar {
        };
    }
}
```

 * with the changes you expected instead:
```php
<?php

class Foo
{
    public function foo($container): void
    {
        $bar = new class(
            $container->get('dependency1'),
            $container->get('dependency2'),
            $container->get('dependency3'),
            $container->get('dependency4'),
            $container->get('dependency5')
        ) extends Bar {
        };
    }
}
```

The [extended coding style guide](https://github.com/php-fig/fig-standards/blob/master/proposed/extended-coding-style-guide.md#8-anonymous-classes) declare that behavior as:
```
Anonymous Classes MUST follow the same guidelines and principles as closures in the above section.
```

### PR DESCRIPTION
**Pull request by jrmajor**
Closes #4289

### CODE REVIEW

File: src/Fixer/ClassNotation/ClassDefinitionFixer.php [hunk 1]
```diff
@@ -150,6 +154,10 @@ protected function createConfigurationDefinition(): FixerConfigurationResolverIn
                 ->setAllowedTypes(['bool'])
                 ->setDefault(false)
                 ->getOption(),
+            (new FixerOptionBuilder('fix_constructor_arguments', 'Whether constructor argument list in anonymous classes should be single line.'))
```
Comment (WinterSilence) [inline src/Fixer/ClassNotation/ClassDefinitionFixer.php:157]: option name doesn't match it's functionality

File: src/RuleSet/Sets/PSR12Set.php [hunk 2]
```diff
@@ -29,7 +29,10 @@ public function getRules(): array
             'braces' => [
                 'allow_single_line_anonymous_class_with_empty_body' => true,
             ],
-            'class_definition' => ['space_before_parenthesis' => true], // defined in PSR12 ¶8. Anonymous Classes
+            'class_definition' => [
+                'fix_constructor_arguments' => false, // handled by method_argument_space fixer
+                'space_before_parenthesis' => true, // defined in PSR12 ¶8. Anonymous Classes
```
Comment (WinterSilence) [inline src/RuleSet/Sets/PSR12Set.php:34]: what is ` ¶8`?

Comment (jrmajor) [reply src/RuleSet/Sets/PSR12Set.php:34]: I think it's supposed to mean “section 8”. From what I understand, it should be `§ 8`, as pilcrow (`¶`) means a paragraph, and what we want is section sign (`§`). [This](https://en.wikipedia.org/wiki/Section_sign#Use) section of Wikipedia article contains some information about the confusion of section and paragraph.

Anyway, keep in mind that it was there before and is not my addition.

File: src/Fixer/ClassNotation/ClassDefinitionFixer.php (hunk 1 above)
Comment (jrmajor) [reply src/Fixer/ClassNotation/ClassDefinitionFixer.php:157]: What would you suggest? Would `single_line_constructor_arguments` be better?

Comment (WinterSilence) [reply src/Fixer/ClassNotation/ClassDefinitionFixer.php:157]: @jrmajor `inline_constructor_arguments`

File: src/RuleSet/Sets/PSR12Set.php (hunk 2 above)
Comment (WinterSilence) [reply src/RuleSet/Sets/PSR12Set.php:34]: @jrmajor you right

**Review by SpacePossum — APPROVED** [review]

**Comment by SpacePossum** [comment]
Thank you @jrmajor.

**Pull request merged**